modules/mining-pipeline/java-maven-project-compiler/src/main/kotlin/org/cafejojo/schaapi/miningpipeline/projectcompiler/javamaven/ProjectCompiler.kt

```
@@ -53,13 +53,11 @@ class ProjectCompiler : ProjectCompiler<JavaMavenProject> {
         }
 
         val result = invoker.execute(request)
+
         if (result.exitCode != 0) {
-            throw ProjectCompilationException("`maven install` executed unsuccessfully: " + result.executionException)
+            logger.warn("`maven install` of ${project.projectDir} failed: ", result.executionException)
+        } else {
+            logger.warn("`maven install` of ${project.projectDir} was successful.")
```

I'm not sure how to deal with situations like these yet. One the one hand, you want to throw an exception. However, this breaks the entire pipeline, and maven install will often fail. However, optionally returning null is rather ugly as this modifies the interface and makes it inconsistent with other interfaces.